engine/runtime/src/main/scala/org/enso/compiler/phase/ImportResolver.scala

```diff
@@ -33,67 +33,107 @@ class ImportResolver(compiler: Compiler) {
   /** Runs the import mapping logic.
     *
     * @param module the entry-point module.
-    * @return a list of all modules that need to be compiled in order to run
-    *         the program.
+    * @return a tuple containing a list of all modules that need to go through the full compilation pipeline and
+    *         a list of all modules which have been inferred from bindings cache and could potentially be compiled lazilly
     */
-  def mapImports(module: Module): List[Module] = {
+  def mapImports(
```

Comment:
Is this the part that is using the new caches to _"speed something up"_?

Reply:
Not much _speeding up_ yet. Well, that's a lie - there is some. We replace calculations of imports/exports in bindings by loading them from the cache. But effectively all modules that were loaded before are loaded now as well.